dexios/src/subcommands/pack.rs

```diff
@@ -19,37 +19,57 @@ use crate::{
     },
 };
 
+pub struct Request<'a> {
+    pub input_file: &'a str,
+    pub output_file: &'a str,
+    pub pack_params: PackParams,
+    pub crypto_params: CryptoParams,
+    pub algorithm: Algorithm,
+}
+
 // this first indexes the input directory
 // once it has the total number of files/folders, it creates a temporary zip file
 // it compresses all of the files into the temporary archive
 // once compressed, it encrypts the zip file
 // it erases the temporary archive afterwards, to stop any residual data from remaining
-#[allow(clippy::module_name_repetitions)]
 #[allow(clippy::too_many_lines)]
-pub fn pack(
-    input: &str,
-    output: &str,
-    pack_params: &PackParams,
-    params: &CryptoParams,
-    algorithm: Algorithm,
-) -> Result<()> {
+pub fn execute(req: Request) -> Result<()> {
     let mut logger = Logger::new();
 
+    // 1. Initialize walker
+    let walker = if req.pack_params.dir_mode == DirectoryMode::Recursive {
+        logger.info(format!("Traversing {} recursively", req.input_file));
+        WalkDir::new(req.input_file)
+    } else {
+        logger.info(format!("Traversing {}", req.input_file));
+        WalkDir::new(req.input_file).max_depth(1)
+    };
+
+    // 2. Skip failed dir entries
+    let walker = walker
+        .into_iter()
+        .filter_map(|res| res.ok())
+        .collect::<Vec<walkdir::DirEntry>>();
+    // let item_data = item.context("Unable to get path of item, skipping")?;
```

Comment:
Dead code that could do with being removed :)

Reply:
I've pushed... but I was too late

Reply:
No worries, I'll remove it in a sec :)